soh/soh/Enhancements/tts/tts.cpp

```diff
@@ -0,0 +1,724 @@
+#include "soh/Enhancements/game-interactor/GameInteractor.h"
+#include "soh/Enhancements/speechsynthesizer/SpeechSynthesizer.h"
+
+#include <OtrFile.h>
+#include <libultraship/classes.h>
+#include <nlohmann/json.hpp>
+#include <spdlog/fmt/fmt.h>
+
+#include "soh/OTRGlobals.h"
+#include "message_data_static.h"
+#include "overlays/gamestates/ovl_file_choose/file_choose.h"
+
+extern "C" {
+extern PlayState* gPlayState;
+}
+
+typedef enum {
+    /* 0x00 */ TEXT_BANK_SCENES,
+    /* 0x01 */ TEXT_BANK_MISC,
+    /* 0x02 */ TEXT_BANK_KALEIDO,
+    /* 0x03 */ TEXT_BANK_FILECHOOSE,
+} TextBank;
+
+nlohmann::json sceneMap = nullptr;
+nlohmann::json miscMap = nullptr;
+nlohmann::json kaleidoMap = nullptr;
+nlohmann::json fileChooseMap = nullptr;
+
+// MARK: - Helpers
+
+std::string GetParameritizedText(std::string key, TextBank bank, const char* arg) {
+    switch (bank) {
+        case TEXT_BANK_SCENES: {
+            return sceneMap[key].get<std::string>();
+            break;
+        }
+        case TEXT_BANK_MISC: {
+            auto value = miscMap[key].get<std::string>();
+            
+            std::string searchString = "$0";
+            size_t index = value.find(searchString);
+            
+            if (index != std::string::npos) {
+                ASSERT(arg != nullptr);
+                value.replace(index, searchString.size(), std::string(arg));
+                return value;
+            } else {
+                return value;
+            }
+            
+            break;
+        }
+        case TEXT_BANK_KALEIDO: {
+            auto value = kaleidoMap[key].get<std::string>();
+            
+            std::string searchString = "$0";
+            size_t index = value.find(searchString);
+            
+            if (index != std::string::npos) {
+                ASSERT(arg != nullptr);
+                value.replace(index, searchString.size(), std::string(arg));
+                return value;
+            } else {
+                return value;
+            }
+            
+            break;
+        }
+        case TEXT_BANK_FILECHOOSE: {
+            return fileChooseMap[key].get<std::string>();
+            break;
+        }
+    }
+}
+
+const char* GetLanguageCode() {
```

Comment:
are we using this in places where we can't use a std::string?

Reply:
mostly it just stood out as different from the rest of the string methods in this file, i'm fine keeping it as `const char*`